public/ledger.js

```diff
@@ -383,28 +383,27 @@ class Ledger {
   /**
    * Send tokens and signatures to ledger before a tx
    *
-   * @param {Object} tokens List of data to send to ledger
-   *   List of data with the format:
+   * @param {Object} tokenMap of data to send to Ledger
+   *   keys are the token uid and values are the token info in the format:
    *     version + uid + len(symbol) + symbol + len(name) + name + len(signature) + signature
    *
    * @return {Promise} Promise resolved when all requests finish
    */
-  sendTokens = async (tokens) => {
+  sendTokens = async (tokenMap) => {
     const values = [];
-    try {
-      const transport = await this.getTransport();
-      for (let [index, data] of tokens.entries()) {
-        const value = await this.sendToLedgerOrQueue(transport, this.commands.SEND_TOKEN, index, 0, data);
-        // Compare response with ledger SW_OK
-        if (ledgerOK.compare(value) !== 0) {
-          // only return failures
-          values.push(data);
-        }
+    const transport = await this.getTransport();
+    let i = 0;
+    for (let [uid, data] of Object.entries(tokenMap)) {
+      try {
+        await this.sendToLedgerOrQueue(transport, this.commands.SEND_TOKEN, i, 0, data);
+      } catch (e) {
+        // only return failures
+        values.push(uid);
```

Comment:
Ledger now is throwing an error instead of returning the not ok value like we had before?
`if (ledgerOK.compare(value) !== 0) {`